modules/nixos/services/bobaboard/default.nix

```diff
@@ -1,16 +1,92 @@
-{ lib, config, ... }:
+{ lib, config, inputs, system, pkgs, ... }:
 
 let
   cfg = config.services.bobaboard;
+  bobabackend-packages = inputs.boba-backend.packages."${system}";
 
   inherit (lib) mkEnableOption mkIf;
 in
 {
   options.services.bobaboard = {
     enable = mkEnableOption "BobaBoard";
   };
-
   config = mkIf cfg.enable {
     # @TODO(jakehamilton): Add configuration for BobaBoard here.
+
+    networking.firewall.allowedTCPPorts = [
+      4200
+    ];
+    
+    users = {
+      users.bobaboard = {
+        isSystemUser = true;
+        group = "bobaboard";
+      };
+
+      groups.bobaboard = {};
+    };
+
+    services.postgresql = {
+      enable = true;
+      package = pkgs.postgresql_12;
+      enableTCPIP = true;
+      authentication = pkgs.lib.mkOverride 10 ''
+        local all all trust
+        host all all 127.0.0.1/32 trust
+        host all all ::1/128 trust
+      '';
+      initialScript = pkgs.writeText "backend-init-script" ''
+        CREATE ROLE the_amazing_bobaboard WITH LOGIN PASSWORD 'how_secure_can_this_db_be' CREATEDB;
+        CREATE DATABASE bobaboard_test;
+        GRANT ALL PRIVILEGES ON DATABASE bobaboard_test TO the_amazing_bobaboard;
+      '';
+    };
+
+
+    services.redis.servers.boba-redis = {
```

Comment:
Maybe `services.redis.servers.bobaboard` instead?

Perhaps even make this configurable.